Fix: Show subtitles on /me/purchases and edit subtitle on /purchases/subscriptions

chrisfromthelc opened the pull request:
#### Changes proposed in this Pull Request

* Update the subtitle from `View, manage, or cancel your WordPress.com plan and other purchases.` to `View, manage, or cancel your plan and other purchases.` to be inclusive of non-WPCOM products (Jetpack, etc.) on `/purchases/subscriptions`. Additionally, add the subtitle to Billing History, "View, print, and email your receipts." and add the subtitle to Payment Methods, "Add or delete payment methods for your account".

<img width="841" alt="Markup 2021-07-21 at 11 36 40" src="https://user-images.githubusercontent.com/8002138/126541801-e55e22e4-1861-4c22-b906-0fd5a8133e0c.png">

* Add the subtitle `View, manage, or cancel your plan and other purchases.` to `/me/purchases` to show on the Active Upgrades tab. Additionally, add the subtitle to Billing History, "View, print, and email your receipts." and add the subtitle to Payment Methods, "Add or delete payment methods for your account".

<img width="795" alt="Screenshot on 2021-07-21 at 11-35-24" src="https://user-images.githubusercontent.com/8002138/126541846-b82c4ec7-a451-4b4e-83b1-90dec691e805.png">

Fixes #52174

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Visit `/purchases/subscriptions` and observe if text is updated to `View, manage, or cancel your WordPress.com plan and other purchases.` on the Active Upgrades tab. Also check for presence of "View, print, and email your receipts." on Billing History and "Add or delete payment methods for your account" on Payment Methods.
* Visit `/me/purchases` and observe if the text `View, manage, or cancel your WordPress.com plan and other purchases.` is present on the Active Upgrades tab. Also check for presence of "View, print, and email your receipts." on Billing History and "Add or delete payment methods for your account" on Payment Methods.

chrisfromthelc committed:
update: text on Upgrades > Purchases

chrisfromthelc committed:
update: text on Me > Purchases

nbloomf reviewed (approved):
Nice; the updated text looks good in both `me/purchases` and `purchases`.

![Screen Shot 2021-07-21 at 1 46 36 PM](https://user-images.githubusercontent.com/9310939/126543220-8fc99d32-605f-4bd6-a1cf-b51a361ce2ec.png)

![Screen Shot 2021-07-21 at 1 46 52 PM](https://user-images.githubusercontent.com/9310939/126543235-6ed06890-e3ce-4358-8d33-8501b4867c81.png)

a8ci18n commented on client/my-sites/purchases/main.tsx:
```diff
@@ -68,9 +68,7 @@ export function Purchases(): JSX.Element {
-				subHeaderText={ translate(
-					'View, manage, or cancel your WordPress.com plan and other purchases.'
-				) }
+				subHeaderText={ translate( 'View, manage, or cancel your plan and other purchases.' ) }
```
:warning: This change will be queued for retranslation. We'll display the English text until then.

michaeldcain replied:
Since this is the site-specific view, can we update it to `View, manage, or cancel your plan and other purchases for this site.`

jjchrisdiehl commented:
Same results as Nathan, this looks good to me:

`/me/purchases`

![image](https://user-images.githubusercontent.com/16580129/126559772-94876a27-feee-436b-8413-b7aa6bae9033.png)

`/purchases/subscriptions/[site]`

![image](https://user-images.githubusercontent.com/16580129/126559698-f6779d19-95da-4beb-92f2-60c3806942b5.png)

Small aside, would we want to just provide copy that is generic such as:

"View, manage, or cancel your ~plan and other~ purchases."

michaeldcain reviewed (changes requested):
The current changes look great, but while we're at it, [can we add subtitles to the other tabs inside the purchase section](https://github.com/Automattic/wp-calypso/issues/52174#issuecomment-864082484)?

<img width="579" alt="Screen Shot 2021-07-21 at 8 42 24 PM" src="https://user-images.githubusercontent.com/942359/126577318-10d7b293-f177-4d16-94d9-d24782b6834c.png">
<img width="531" alt="Screen Shot 2021-07-21 at 8 42 29 PM" src="https://user-images.githubusercontent.com/942359/126576782-34c79837-637d-4ffb-a34b-cdfc1c8c7590.png">

I think for Billing History we could say "View, print, and email your receipts." and for Payment Methods, "Add or delete payment methods for your account". These would persist to the sub-pages of the tabs. We can do this in a follow-up PR, but in that case, remove the "Fixes #---" from this PR.

chrisfromthelc committed:
update: additional text on Me > Purchases and Upgrades > Purchases for Billing History and Payment Methods tabs

chrisfromthelc commented:
> The current changes look great, but while we're at it, [can we add subtitles to the other tabs inside the purchase section](https://github.com/Automattic/wp-calypso/issues/52174#issuecomment-864082484)?
>  
> I think for Billing History we could say "View, print, and email your receipts." and for Payment Methods, "Add or delete payment methods for your account". These would persist to the sub-pages of the tabs. We can do this in a follow-up PR, but in that case, remove the "Fixes #---" from this PR.

Sorry, I misread that section to mean that those tabs should specifically not have anything. I've added the suggested text on Billing History and Payment Methods in both `/me/purchases` and `/purchases/subscriptions`.

One thing I wanted to note is that in [client/me/purchases/payment-methods/main.tsx](https://github.com/Automattic/wp-calypso/pull/54772/files#diff-59f41ee55370a1db7bbbd207b3fefb26df80126dec8b05953035d6d08b3eab06), `useTranslate()` was not being imported. I went ahead and imported that in line with how the other sections were imported in their respective files.

The sections should render now as shown in these screenshots.

<img width="474" alt="Screenshot on 2021-07-21 at 21-21-15" src="https://user-images.githubusercontent.com/8002138/126590805-a0f3a3aa-001e-4c63-91d1-b568ccd1a9c8.png">
<img width="480" alt="Screenshot on 2021-07-21 at 21-21-25" src="https://user-images.githubusercontent.com/8002138/126590821-c694c1d0-ee4e-4c64-babf-7a4546ab8ac2.png">
<img width="478" alt="Screenshot on 2021-07-21 at 21-21-33" src="https://user-images.githubusercontent.com/8002138/126590832-85baaa1c-efcf-437f-848a-e0b3e3604486.png">

I've also updated the original PR description to include the addition of the Billing History and Payment Methods subtitles.

michaeldcain commented on client/my-sites/purchases/billing-history/index.tsx:
```diff
@@ -76,6 +76,7 @@ export function BillingHistory( { siteSlug }: { siteSlug: string } ) {
 				brandFont
 				className="billing-history__page-heading"
 				headerText={ titles.sectionTitle }
+				subHeaderText={ translate( 'View, print, and email your receipts.' ) }
```
Since this is the site-specific view, can we update it to `View, print, and email your receipts for this site.`

chrisfromthelc replied:
Instead of `View, print, and email your receipts for this site.`, should we go as far as `View, print, and email your receipts for example.com` with the specific site address (and replicate to the other site-specific views)?

michaeldcain replied:
I think that's a good idea, but I'm worried about any confusion that might be attached to multiple domains on a single site, or for instance if an email was for a secondary domain. Customer Home and Domains both use "your site", which I thought about doing, but "email _your_ receipts for _your_ site" felt redundant.

chrisfromthelc replied:
This is now added.

<img width="490" alt="Screen Shot 2021-07-23 at 12 33 12 PM" src="https://user-images.githubusercontent.com/8002138/126832502-648f753e-c7a8-4085-a4a7-6cc633cc5266.png">

Marking as resolved.

michaeldcain commented on client/my-sites/purchases/payment-methods/index.tsx:
```diff
@@ -72,6 +72,7 @@ export function PaymentMethods( { siteSlug }: { siteSlug: string } ): JSX.Elemen
 				brandFont
 				className="payment-methods__page-heading"
 				headerText={ titles.sectionTitle }
+				subHeaderText={ translate( 'Add or delete payment methods for your account.' ) }
```
This one is good - even though it's the site-specific view, payment methods are managed for the entire account.

chrisfromthelc replied:
`client/my-sites/purchases/main.tsx` is corrected now as well.

<img width="530" alt="Screen Shot 2021-07-23 at 12 32 51 PM" src="https://user-images.githubusercontent.com/8002138/126832602-e843d196-4693-476e-b8ff-677c2e7a0590.png">

Marking as resolved.

michaeldcain reviewed (changes requested):
This is looking good. In testing, I noticed that it's probably worth adding something to indicate that some of these views are site-specific. Sorry for the additional back and forth on this one.

chrisfromthelc committed:
Added 'for this site' for /purchases/subscriptions

a8ci18n commented on client/me/purchases/purchases-list/index.jsx:
```diff
@@ -177,7 +177,12 @@ class PurchasesList extends Component {
+				<FormattedHeader
+					brandFont
+					headerText={ titles.sectionTitle }
+					subHeaderText={ translate( 'View, manage, or cancel your plan and other purchases.' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'View, manage, or cancel your WordPress.com plan and other purchases.' )` *ES Score: 10*

chrisfromthelc replied:
Resolving since this is a new string being added intentionally.

a8ci18n commented on client/my-sites/purchases/main.tsx:
```diff
@@ -69,7 +69,7 @@ export function Purchases(): JSX.Element {
 				headerText={ titles.sectionTitle }
 				subHeaderText={ translate(
-					'View, manage, or cancel your WordPress.com plan and other purchases.'
+					'View, manage, or cancel your plan and other purchases for this site.'
```
:warning: This change will be queued for retranslation. We'll display the English text until then.

chrisfromthelc commented:
The E2E tests for `[WPCOM] Editor: Media Upload (mobile) @parallel @jetpack: Can upload a normal image` are continually failing, and appear unrelated to the changes in this PR. It looks as if it's timing out in trying to select the Image block.

After chatting with @sirbrillig, it seems that it'll be okay to ignore and merge this.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6277384

Thank you @chrisfromthelc for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6277384).
